### PR TITLE
fix issue 628 and change oqs_test_evp_pkey_params.c to use TEST_ASSERT()

### PR DESCRIPTION
**Pull request by HelenHZhang**

### CODE REVIEW

**Comment by HelenHZhang** [comment]
> > I use the command in issue 628 to recreate the problem and test the fix using the same command.
> 
> Understood. Since we currently don't have (AFAIK) any testing mechanism for context strings, it would be great if you could add, for example, a sub-test within `oqs_test_signatures` to include the use of a (non-null) context string of your choice, so we can have this tested.
> 
> Any other way to contribute a test of this fixed behavior would be much appreciated.

I read oqs_test_signatures.c::test_oqs_signatures()
The first section (line 41-60) and 2nd section (line 81 - 95, including line 72-79) are essentially the same, except the first section using "SHA512" and 2nd section using the built-in digest algorithm.
I would think the clean-up needs to be done in this file.

The first section only runs in "default" provider, which means both sections runs in "default" provider.
In other words, the tests covers explicit "SHA512" and built-in default digest in default provider.
The tests cover only built-in digest in FIPS provider.
-- is this the intention for test design?

Since I am new to oqs-provider, I don't want to make too many changes in this simple PR.

**Comment by RodriM11** [comment]
@HelenHZhang since it is not currently possible to add automatic testing, as some algorithms will fail (not all support context strings) and it would require to introduce a list of algorithms to actually test, I have to try manually to be able to confirm that the error is indeed resolved. I will do it this week and approve if so. I apologize for the delay.

**Comment by RodriM11** [comment]
I have now checked the solution and indeed it (logically) works and the double free is no longer there. Therefore, the PR LGTM! (I understand the change to made to the `evp_pkey_params` test is just to homogenize return values among all tests, right?) Thank you very much for your contribution and I apologize for the delay in my response.

Unfortunately, I don't have "official" approval "powers" yet, but you have my approval nonetheless. Ping to @baentsch to see if he can also revise and approve if he considers it so.

File: test/oqs_test_evp_pkey_params.c [hunk 1]
```diff
@@ -400,7 +400,7 @@ int main(int argc, char **argv) {
     const OSSL_ALGORITHM *algs;
     int query_nocache;
     int errcnt;
-    int ret = EXIT_FAILURE;
+    int test = 0;
```
Comment (baentsch) [inline test/oqs_test_evp_pkey_params.c]: What's the reason for this variable? It looks like it's never changed: Can it be removed for better code readability?

**Review by baentsch — COMMENTED** [review]
Thanks for checking the logic of this @RodriM11 . I thus now only have a "code understanding nit" before I'd approve. AFAIK and FWIW, @bhess, @ashman-p, @dstebila @xuganyu96 can always approve anything in this project independent of my availability.
